### PR TITLE
New outport driver interface - take 2

### PR DESCRIPTION
**Pull request by iulianpascalau**
- new outport driver interface
- made the outport handler wait for all drivers to return nil error on each function

### CODE REVIEW

**Review by liviuancas-elrond — APPROVED** [review]
test passed :+1: 
steps:

node is in sync
external service goes down
node falls behind the chain
external service is back up and responds correctly to pushed events
node gets back in sync

**Review by SebastianMarian — APPROVED** [review]

File: outport/outport.go [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
-var log = logger.GetOrCreate("outport")
-
 // NewOutport will create a new instance of proxy
-func NewOutport() *outport {
-	return &outport{
-		drivers: make([]Driver, 0),
-		mutex:   sync.RWMutex{},
+func NewOutport(retrialInterval time.Duration) (*outport, error) {
+	if retrialInterval < minimumRetrialInterval {
+		return nil, fmt.Errorf("%w, provided: %d, minimum: %d", ErrInvalidRetrialInterval, retrialInterval, minimumRetrialInterval)
```
Comment (AdoAdoAdo) [inline outport/outport.go:28]: :+1:  
we can avoid high(100%) cpu load if flag is missing

**Review by AdoAdoAdo — APPROVED** [review]

File: outport/outport.go (hunk 1 above)
Comment (iulianpascalau) [reply outport/outport.go:28]: exactly. We should always have minimums when doing endless fors.

**Review by gabi-vuls — APPROVED** [review]
System test passed
image

**Pull request merged**